### ISSUE TITLE
Periodic freezing during WYSIWYG text-entry into very long note, under Linux (lesser extent in Windows)

### ISSUE THREAD

**Issue by kauffy** [laurent22/joplin#5167]
## Environment

Joplin version: 2.18
Platform: Linux Mint
OS specifics: 20.1

## Steps to reproduce

1. Have a very long note (~70K words, with 1 single, small image near the top).
2. Continue to type in the note in WYSIWYG.
3. Observe: at least once a line (at least a few times in any given 5 minute period), Joplin will appear to freeze, the caret will stop flashing, and no text will appear when typed.
4. After a varying-length interval (sometimes, as long as five or ten seconds-- I counted roughly 6 seconds a few times), any keystrokes pressed during the freeze will be fired off when it unfreezes.

## Describe what you expected to happen

Textual input would be continuous and smooth, as it is in Evernote or Note Station (or Libre Writer, et al.)

Some additional notes about my note:
Almost the entire thing is a 100 page bullet-list (it's just a writing approach I use). There are some hard line-rules inserted at roughly chapter-type intervals. Other than that, it's just text, with the only formatting being bold or italics.
This file had more images in it before, but I was experiencing this issue severely, so I tried moving the images to a separate note, which did seem to improve performance. The performance is degrading again.

It also appears that I get this issue under Windows, but my Windows machine is a 4.5Ghz i7 with 32GB of RAM, v. the Linux laptop that is 2.5GHz i3 with 12GB of RAM.

## Logfile

(I will have to gather logs-- I didn't realize they needed to be turned on, so I don't have any. Tonight, I will enable them before I do my writing, and I can attach them if this issue is of interest. I will also try to differentiate the Markdown v. WYSIWYG editors.)

**Comment by kauffy** [comment]
Commenting to add:
This does not appear to be related to synchronization at all. Sync is happening about every two minutes, and when this freezing occurs, I will look over at the sync information to see when it last sync'd and it isn't proximal.

**Comment by leematos** [comment]
when you say WYSIWYG, Markdown or Rich Text? Any plugins and/or what markdown settings do you have? Some of those could be heavy, and depending on how the text is parsed and rendered that flow could be the root here.

**Comment by kauffy** [comment]
The WYSIWYG is the Rich Text-- not Markdown. I'm unsure if Markdown does the same thing.

I _was_ thinking that it could be a plugin, as I've installed several-- it looks like there are two that are suspect, maybe three, the rest are things that must be activated. Math Mode and Inline Tags, and possibly Rich Markdown.

Embarrassingly, I didn't even think of the plugins as something to test before I posted. I'll try this and post back.
